### PR TITLE
chore(types): type check the rest of the protocol

### PR DESCRIPTION
**Pull request by JoelEinbinder**
This adds events and commands into `protocol.d.ts`

### CODE REVIEW

File: lib/EmulationManager.js [hunk 1]
```diff
@@ -33,13 +33,13 @@ class EmulationManager {
...
     const height = viewport.height;
     const deviceScaleFactor = viewport.deviceScaleFactor || 1;
+    /** @type {Protocol.Emulation.ScreenOrientation} */
     const screenOrientation = viewport.isLandscape ? { angle: 90, type: 'landscapePrimary' } : { angle: 0, type: 'portraitPrimary' };
 
     await Promise.all([
       this._client.send('Emulation.setDeviceMetricsOverride', { mobile, width, height, deviceScaleFactor, screenOrientation }),
       this._client.send('Emulation.setTouchEmulationEnabled', {
-        enabled: viewport.hasTouch || false,
-        configuration: viewport.isMobile ? 'mobile' : 'desktop'
```
Comment (aslushnikov) [inline lib/EmulationManager.js:42]: why is `configuration` removed?

File: lib/externs.d.ts [hunk 2]
```diff
@@ -12,7 +12,10 @@ import * as child_process from 'child_process';
 declare global {
   module Puppeteer {
     export class Connection extends RealConnection {}
-    export class CDPSession extends RealCDPSession {}
+    export class CDPSession extends RealCDPSession {
+      on<T extends keyof Protocol.Events>(event: T, listener: (arg: Protocol.Events[T]) => void): this;
```
Comment (aslushnikov) [inline lib/externs.d.ts:16]: this looks like black magic

File: lib/EmulationManager.js (hunk 1 above)
Comment (JoelEinbinder) [reply lib/EmulationManager.js:42]: There is no such option on `Emulation.setTouchEmulationEnabled`. It is only on `Page.setTouchEmulationEnabled`.

File: lib/externs.d.ts (hunk 2 above)
Comment (JoelEinbinder) [reply lib/externs.d.ts:16]: It is!

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**